### ISSUE TITLE
R: fails on aarch64-darwin with BLAS compile error

### ISSUE THREAD

**Issue by ConnorBaker** [NixOS/nixpkgs#158730]
### Describe the bug

I'm unable to install or use R on `aarch64-darwin` via `nix build` or `nix shell`. When I try either of those commands, I see errors like the below:

```text
/private/tmp/nix-build-R-4.1.2.drv-2/cceZuOGJ.s:22371:15: error: index must be an integer in range [-256, 255].
        ldr     x8, [x8, ___stack_chk_guard];momd
                         ^
/private/tmp/nix-build-R-4.1.2.drv-2/cceZuOGJ.s:22435:15: error: index must be an integer in range [-256, 255].
        ldr     x0, [x0, ___stack_chk_guard];momd
                         ^
make[4]: *** [../../../Makeconf:130: blas.o] Error 1
make[4]: *** Waiting for unfinished jobs....
/private/tmp/nix-build-R-4.1.2.drv-2/ccT4z6Ab.s:1154:15: error: index must be an integer in range [-256, 255].
        ldr     x8, [x8, ___stack_chk_guard];momd
                         ^
/private/tmp/nix-build-R-4.1.2.drv-2/ccT4z6Ab.s:1505:15: error: index must be an integer in range [-256, 255].
        ldr     x0, [x0, ___stack_chk_guard];momd
                         ^
```

 fails on `aarch64-darwin`

### Steps To Reproduce

```bash
nix shell nixpkgs#R
```

### Expected behavior

R should be available in the shell launched by `nix shell nixpkgs#R`.

### Additional context

You can find a log of the attempt at running `nix shell` here: <https://gist.github.com/ConnorBaker/e3e33aa61f3269e18cd2854ff69508c1>

I was reading through CRAN's webpage for R and noticed on [the page for macOS builds](https://cran.r-project.org/bin/macosx/):

> If you wish to compile R packages from sources, you may need to download GNU Fortran for arm64 from https://mac.r-project.org/libs-arm64/. Any external libraries and tools are expected to live in /opt/R/arm64 to not conflict with Intel-based software and this build will not use /usr/local to avoid such conflicts.

Could this error I'm seeing be related to not using the `arm64` components described there?

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
% nix-shell -p nix-info --run "nix-info -m"
these 6 paths will be fetched (0.55 MiB download, 2.81 MiB unpacked):
  /nix/store/0mr7pcp32apiipy2chwbm40xiyr4rwk2-bash-interactive-5.1-p12-man
  /nix/store/1vf7mgrsymraipkqf0pf1m0629lzxdkc-nix-info
  /nix/store/3q0fx8ndv285rjbjj4zpar74v2j1hmd3-bash-interactive-5.1-p12-doc
  /nix/store/4fkx7bxi4sl2aqcr8j23v1w9c3dh73xa-bash-interactive-5.1-p12-info
  /nix/store/d5wh3wv76ylwzm3a492g0wlcdb344rgq-bash-interactive-5.1-p12-dev
  /nix/store/rqv2pkd4pj3c5r0154bl066j55g2xj6z-DarwinTools-1
copying path '/nix/store/4fkx7bxi4sl2aqcr8j23v1w9c3dh73xa-bash-interactive-5.1-p12-info' from 'https://cache.nixos.org'...
copying path '/nix/store/3q0fx8ndv285rjbjj4zpar74v2j1hmd3-bash-interactive-5.1-p12-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/0mr7pcp32apiipy2chwbm40xiyr4rwk2-bash-interactive-5.1-p12-man' from 'https://cache.nixos.org'...
copying path '/nix/store/d5wh3wv76ylwzm3a492g0wlcdb344rgq-bash-interactive-5.1-p12-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/rqv2pkd4pj3c5r0154bl066j55g2xj6z-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/1vf7mgrsymraipkqf0pf1m0629lzxdkc-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.3.0, macOS 12.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.7.0pre20220127_558c4ee`
 - channels(connorbaker): `"darwin, nixpkgs"`
 - channels(root): `"nixpkgs-22.05pre340045.f01adc7b35a"`
 - nixpkgs: `/Users/connorbaker/.nix-defexpr/channels/nixpkgs`
```

**Comment by ConnorBaker** [comment]
@timokau, @omasanori, @collares, @jbedo, and raskin, you all are listed as maintainers, apologies if I'm not supposed to tag you.

Have any of you seen this behavior or have any ideas of any easy fix besides special-casing `aarch64-darwin` and downloading the pre-built? I'm not familiar enough with what linking looks like on macOS to know what I'd need to change to get things working with the arm64 files they provide.

**Comment by smaret** [comment]
This should have been fixed by #158992.
